src/pages/index.js

```diff
@@ -1,90 +1,233 @@
+);
+const profileImageEditButton = document.querySelector(
+  ".profile-image__edit-button"
+);
 
-// Function to create a card
+// API Initialization
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "740df34c-ea9e-4ff2-ad05-85becca779ab",
+    "Content-Type": "application/json",
+  },
+});
+
+let cardList;
+let userInfo;
+
+// Load Initial Data
+api
+  .loadPageResults()
+  .then(([card, userData]) => {
+    // Initialize Card List
+    cardList = new Section(
+      {
+        items: [...initialCards, ...card],
+        renderer: createCard,
+      },
+      ".gallery__cards"
+    );
+    cardList.renderItems();
+
+    // Initialize User Info
+    userInfo = new UserInfo(
+      ".profile__title",
+      ".profile__description",
+      ".profile__image"
+    );
+    userInfo.setUserInfo({ title: userData.name, description: userData.about });
+    userInfo.setUserImage(userData.avatar);
+  })
+  .catch(console.error);
+
+// Function to Create Card
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handlePreviewPicture,
+    handleDeleteClick,
+    handleLikeClick
+  );
+  return cardElement.getView();
 }
 
-// Initialize Cards Section
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".gallery__cards"
-);
-
-cardSection.renderItems();
+// Profile Edit Submit Handler
+function handleProfileEditSubmit(userData) {
+  profilePopupForm.setLoading(true);
+  api
+    .fetchEditProfile(userData)
+    .then(() => {
+      userInfo.setUserInfo({
+        title: userData.title,
+        description: userData.description,
+      });
+      profilePopupForm.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => profilePopupForm.setLoading(false));
+}
 
-// Initialize Popups
-const imagePopup = new PopupWithImage("#preview-image-modal");
-imagePopup.setEventListeners();
+// Initialize Profile Image Popup Form
+const profileImagePopupForm = new PopupWithForm(
+  "#picture-modal",
+  handleProfilePictureEdit
+);
+profileImagePopupForm.setEventListeners();
 
-const addCardForm = document.querySelector("#add-card-form");
-const addCardPopup = new PopupWithForm("#add-card-modal", (formData) => {
-  const cardData = { name: formData.title, link: formData.url };
-  const cardElement = createCard(cardData);
-  cardSection.addItem(cardElement);
-  addCardForm.reset();
-  addCardFormValidator.resetValidation();
-  addCardPopup.close();
+// Profile Image Edit Button Click Event
+profileImageEditButton.addEventListener("click", (evt) => {
+  evt.preventDefault();
+  profileImagePopupForm.open();
+  addFormValidator.disableButton();
+  avatarFormElement.reset();
```

Comment:
You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

And `.disableButton();` should be also in `then`

Reply:
Hello! I've tried to figure this out but I'm still confused. This is how I understand it. Putting the reset inside the `then` block will complete the promise, while it may never happen if it's outside of the `then` block. I just don't know how to add it to this form. Please advise.